edge/pkg/devicetwin/dtmanager/twin_test.go

```diff
@@ -0,0 +1,1534 @@
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeFunc(content),
+			errWant: errors.New("Update twin error, the update request body not have key:twin"),
+		},
+		{
+			name:     "TestDealTwinSync(): Case 4: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgTypeFunc(contentKeyTwin),
+			errWant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinSync(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealTwinSync() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealTwinGet is function to test dealTwinGet
+func TestDealTwinGet(t *testing.T) {
+	contentKeyTwin, _ := json.Marshal(keyTwinUpdateFunc())
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		errWant  error
+	}{
+		{
+			name:    "TestDealTwinGet(): Case 1: msg not Message type",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeErrorFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("msg not Message type"),
+		},
+		{
+			name:    "TestDealTwinGet(): Case 2: invalid message content",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinGet(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgTypeFunc(contentKeyTwin),
+			errWant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinGet(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealTwinGet() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealTwinUpdate is function to test dealTwinUpdate
+func TestDealTwinUpdate(t *testing.T) {
+	content, _ := json.Marshal(dttype.DeviceTwinUpdate{BaseMessage: dttype.BaseMessage{EventID: "Event1"}})
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		errWant  error
+	}{
+		{
+			name:    "TestDealTwinUpdate(): Case 1: msg not Message type",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeErrorFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("msg not Message type"),
+		},
+		{
+			name:    "TestDealTwinUpdate(): Case 2: invalid message content",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinUpdate(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgTypeFunc(content),
+			errWant:  nil,
```

Comment:
we need to add tests where 
- the device twin update is rejected if twin is not found , 
- the update request body doesn't contain the keyword "twin"